packages/compiler-sfc/src/compileScript.ts

```diff
@@ -130,6 +130,10 @@ export interface SFCScriptCompileOptions {
    * using it, disable this and switch to the [Vue Macros implementation](https://vue-macros.sxzz.moe/features/reactivity-transform.html).
    */
   reactivityTransform?: boolean
+  /**
+   * Transform Vue SFCs into custom elements.
+   */
+  customElement?: boolean | ((filename: string) => boolean)
```

Comment:
LGTM, but I'm still considering if we need `boolean` type. There's seldom a change to pass a boolean value to make all component is or isn't a custom element.

Maybe we can simplify it a bit further.

Reply:
Because I consider that a bool variable `asCustomElement` is maintained at the beginning of the whole logic in `vite-plugin-vue`, so the customElement option supports Boolean in vue/core

Reply:
Fair enough